basic_hashtable/b_hashtables.c

```diff
@@ -94,7 +109,17 @@ void hash_table_insert(BasicHashTable *ht, char *key, char *value)
  ****/
 void hash_table_remove(BasicHashTable *ht, char *key)
 {
+  unsigned int index = hash(key, ht->capacity);
 
+  if (ht->storage[index] == NULL)
+  {
+    printf("Nonexistent key: %s", key);
+    exit(1);
+  }
+  else
+  {
+    destroy_pair(ht->storage[index]);
```

Comment:
it's not enough to `destroy_pair`, `ht->storage[index]` is still going to have junk
you need to `ht->storage[index] = NULL;` as well